cds_sorenson/api.py

```diff
@@ -43,78 +44,51 @@ def _str_(self):
-    output['QueueId'] = current_app.config['CDS_SORENSON_DEFAULT_QUEUE']
-
-    source_media = {}
-    source_media['FileUri'] = input_folder + input_filename
-    source_media['UserName'] = current_app.config['CDS_SORENSON_USERNAME']
-    source_media['Password'] = current_app.config['CDS_SORENSON_PASSWORD']
-
-    jobInfo = {}
-    jobInfo['SourceMediaList'] = [source_media]
-
-    destination_list = {}
-    destination_list['FileUri'] = output_folder + output_filename
-    jobInfo['DestinationList'] = [destination_list]
-
-    preset_id_json = {"PresetId": preset_id}
-    jobInfo['CompressionPresetList'] = [preset_id_json]
-
-    output['JobMediaInfo'] = jobInfo
-
-    return output
-
-
-def start_encoding(input_filename, preset_name):
+def start_encoding(input_file, preset_name):
     """Encode a video that is already in the input folder.
 
-    :param filename: string with the filename.
+    :param filename: string with the filename, something like
+        /eos/cds/test/sorenson/8f/m2/728-jsod98-8s9df2-89fg-lksdjf/data where
+        the last part "data" is the filename and the last directory is the
+        bucket id.
     :param preset: id of the preset (taken from sorenson dashboard).
     :returns: job ID.
     """
     current_app.logger.debug("Encoding {0} with the preset {1}"
-                             .format(input_filename, preset_name))
-
-    preset_config = current_app.config['CDS_SORENSON_PRESETS'].get(preset_name)
-    preset_id, extension = preset_config
-
-    # Output file extension depends on the selected preset
-    output_basename = os.path.splitext(input_filename)[0]
-    output_filename = output_basename + extension
+                             .format(input_file, preset_name))
 
     # Build the request of the encoding job
-    json_params = _generate_json(input_filename, output_filename, preset_id)
+    json_params = generate_json_for_encoding(input_file, preset_name)
 
-    current_app.logger.debug("Sending request to the Sorenson server")
     headers = {'Accept': 'application/json'}
     response = requests.post(current_app.config['CDS_SORENSON_SUBMIT_URL'],
                              headers=headers, json=json_params)
 
     data = json.loads(response.text)
 
-    current_app.logger.debug("Response from Sorenson: {}".format(data))
     if response.status_code == requests.codes.ok:
         job_id = data.get('JobId')
         return job_id
     else:
-        raise SorensonError(
-            "Failed to send encoding request to the server. Received "
-            "code: {0}. Make sure all parameters are set and permissions are "
-            "correct".format(response.status_code)
-        )
+        # something is wrong - sorenson server is not responding or the
+        # configuration is wrong and we can't contact sorenson server
+        raise SorensonError(response.status_code)
+
+
+def batch_start_encoding(input_files, presets_names):
```

Comment:
I don't see the need of sending several files to transcode at once, our use case goes more to given **one** master file generate **several** slaves using a given list of presets.

What about changing the API to something like:

```python
def batch_start_encoding(input_file, *presets):
```